tiledb/sm/filesystem/gcs.h

```diff
@@ -35,16 +35,7 @@
 
 #ifdef HAVE_GCS
 
-#if defined(_MSC_VER)
-#pragma warning(push)
-// One abseil file has a warning that fails on Windows when compiling with
-// warnings as errors.
-#pragma warning(disable : 4127)  // conditional expression is constant
-#endif
-#include <google/cloud/storage/client.h>
-#if defined(_MSC_VER)
-#pragma warning(pop)
-#endif
+#include <google/cloud/version.h>
```

Comment:
We have to include this header because the GCS client class is declared in an inline namespace, but it's much smaller.